modules/cli/src/main/scala/scala/cli/commands/package0/Package.scala

```diff
@@ -101,12 +118,15 @@ object Package extends ScalaCommand[PackageOptions] with BuildCommandHelpers {
       ) { res =>
         res.orReport(logger).map(_.main).foreach {
           case s: Build.Successful =>
-            s.copyOutput(options.shared)
             val mtimeDestPath = doPackage(
               logger = logger,
               outputOpt = options.output.filter(_.nonEmpty),
               force = options.force,
-              forcedPackageTypeOpt = options.forcedPackageTypeOpt,
+              forcedPackageTypeOpt =
+                options.forcedPackageTypeOpt orElse detectNativePackageType(
+                  s.options.platform.value,
+                  s.options.scalaNativeOptions
+                ),
```

Comment:
I think that instead of executing `detectNativePackageType` here, it can be handled in this [section](https://github.com/VirtusLab/scala-cli/blob/d062ea14f3286adbad722117bbac45cb69284085/modules/cli/src/main/scala/scala/cli/commands/package0/Package.scala#L222)